src/client/src/Stream.c

```diff
@@ -516,28 +539,31 @@ STATUS stopStream(PKinesisVideoStream pKinesisVideoStream)
         sessionCount == 0 && // If we have active handle, then eventually one of the handle will call streamClosedFn
         !pKinesisVideoStream->metadataTracker.send &&
         !pKinesisVideoStream->eosTracker.send) {
+        DLOGW("Calling notifyStreamClosed...");
         CHK_STATUS(notifyStreamClosed(pKinesisVideoStream, pUploadHandleInfo == NULL ?
                                       INVALID_UPLOAD_HANDLE_VALUE : pUploadHandleInfo->handle));
+        DLOGW("Finished Calling notifyStreamClosed...");
+
     }
 
+    // Unlock the stream (even though it will be unlocked in the cleanup)
```

Comment:
No need for this change

Reply:
fixes typo where ) was missing

I just realized I introduced a bug here by reversing the order in which the mutexes are freed (this was left over from an earlier version when i was playing with reversing the order of acquiring the mutexes) I'm reverting that

Reply:
BTW, could I ask you to resolve these convos once you fix them/resolve or any other-wise so we could keep the track